source/docs/advanced/theme/analytics.md

```diff
@@ -0,0 +1,237 @@
+
+## Track an event
+
+An event is something that happens in your application. For instance, it can happens when a user clicks on a button, opens a dropdown, etc. It usually conveys meaning to your marketing team to better understand what drives your users.
+
+Most of the e-commerce related events are already implemented within Front-Commerce. But each website will have different behaviors, and it can be interesting to add your own events to see how your customers uses your application.
+
+To do so, you will need to call the method `trackEvent` from `web/utils/analytics`.
+
+For instance, let's say that you are building a grocery store and that you have created Recipe pages that display displays at the end the list of ingredients needed for the current recipe. Below this list, you have a created a button that adds all the ingredients to the cart of the user, and you want to know if this button is useful and if the users click on it.
+
+The button would likely be a component that would look like this:
+
+```jsx
+import React from "react";
+import Button from "theme/components/atoms/Button";
+
+const AddIngredientsToCart = ({addToCart}) => {
+  return (
+    <Button onClick={() => {
+      addToCart();
+    }}>
+      Add ingredients
+    </Button>
+  );
+}
+
+export default AddIngredientsToCart;
+```
+
+To add your tracking, you would need to call the `trackEvent` method when the user clicks on the Button. Thus, your new component would look like this:
+
+```diff
+import React from "react";
+import Button from "theme/components/atoms/Button";
++import { trackEvent } from "web/core/analytics";
+
+const AddIngredientsToCart = ({addToCart, ingredients}) => {
+  return (
+    <Button onClick={() => {
++     trackEvent("Added ingredients to cart", {
++       ingredients: ingredients
++     });
+      addToCart();
+    }}>
+      Add ingredients
+    </Button>
+  );
+}
+
+export default AddIngredientsToCart;
+```
+
+This `trackEvent` method is actually a short cut that lets you call the `analytics.track` method of `analytics.js`. It uses the exact same API. See [the document] for more detailed information.
+
+Additionally, you may wonder what name and properties you should give to your events. It depends on the integrations you are using and if you want to create a new event or use Semantic ones (supported events within Segment.io). To learn more about this, please refere to the [Semantic Events documentation](https://segment.com/docs/spec/semantic/).
+
+### Track an event as a React Component
+
+If you don't have an actual callback where to put the `trackEvent` (like `onClick`), you can use the `withTrackOnMount` enhancer that will let you call the `trackEvent` using [React lifecycle]((http://projects.wojtekmaj.pl/react-lifecycle-methods-diagram/)).
+
+```jsx
+import withTrackOnMount from "theme/modules/Analytics/withTrackOnMount";
+
+withTrackOnMount({
```

Comment:
It would be great to extract this in the Reference section (along with the `<TrackOnMount />` component) and prefer an example here. We could use an example from the core for instance.